### DISCUSSION TITLE
How can you use parameters with spaces in values?

### DISCUSSION THREAD

**Discussion by brendan-stephens** [knqyf263/pet#270] [Q&A] [answered]
--It doesn't seem like `<param ...>` is read at the end of an flag, making it difficult to supply default values.--

Edit: issue appears quoted string support.

Eg. `--user=<param='some name'>`

**Comment by brendan-stephens** [comment]
```
example_function() {
    local flag_argument

    # Parse command line options
    while [[ $# -gt 0 ]]; do
        case $1 in
            --flag=*)
                flag_argument="${1#*=}"
                ;;
            --flag)
                shift
                flag_argument="$1"
                ;;
            *)
                echo "Invalid option: $1" >&2
                return 1
                ;;
        esac
        shift
    done

    # Main function logic
    if [[ -n $flag_argument ]]; then
        echo "Flag argument: $flag_argument"
    else
        echo "Flag argument not provided."
    fi
 }
 
❯ example_function --flag=value
Flag argument: value
```

```
[[snippets]]
  description = "flags"
  command = "example_function --flag=<param='My Value'>"
  output = ""

❯ pet exec
sh: -c: line 0: syntax error near unexpected token `newline'
sh: -c: line 0: `example_function --flag=<param='My Value'>'
exit status 2
```

**Comment by brendan-stephens** [comment]
Actually, I think the issue is support for quoted variables:

`example_function --flag=<param=Bananas>` works
`example_function --flag=<param='Bananas'>` works
`example_function --flag=<param=Lots of Bananas>` fails
`example_function --flag=<param='Lots of Bananas'>` fails

**Comment by RamiAwar** [comment]
Thanks for reporting this and adding these test cases! Will fix this up for you asap :)

**Comment by RamiAwar** [comment]
So it seems like the regex pattern doesn't allow for empty spaces in the values right now, which is in line with all your examples above. This should be an easy fix

<img width="448" alt="image" src="https://github.com/knqyf263/pet/assets/8523191/b1e6418f-b8fc-4254-98ff-67b98bb7556c">

**Comment by RamiAwar** [comment, marked as answer]
Just added support for this, will be in the next release tonight/tomorrow! @brendan-stephens
